src/Mod/Path/PathScripts/post/nccad_post.py

```diff
@@ -0,0 +1,116 @@
+# *   it under the terms of the GNU Lesser General Public License (LGPL)                      *
+# *   as published by the Free Software Foundation; either version 2 of                       *
+# *   the License, or (at your option) any later version.                                     *
+# *   for detail see the LICENCE text file.                                                   *
+# *                                                                                           *
+# *   FreeCAD is distributed in the hope that it will be useful,                              *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of                          *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the                           *
+# *   GNU Lesser General Public License for more details.                                     *
+# *                                                                                           *
+# *   You should have received a copy of the GNU Library General Public                       *
+# *   License along with FreeCAD; if not, write to the Free Software                          *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307                    *
+# *   USA                                                                                     *
+# *                                                                                           *
+# ********************************************************************************************/
+import FreeCAD
+import PathScripts
+from PathScripts import PostUtils
+import datetime
+
+
+TOOLTIP='''
+This is a postprocessor file for the Path workbench. It is used to take
+a pseudo-gcode fragment outputted by a Path object and output real GCode
+suitable for the Max Computer GmbH nccad9 Computer Numeric Control.
+
+Supported features:
+
+- 3-axis milling
+- manual tool change with tool number as comment
+- spindle speed as comment
+
+!!! gCode files must use the suffix .knc !!!'''
+
+
+MACHINE_NAME = '''Max Computer GmbH nccad9 MCS/KOSY'''
+
+
+# gCode for changing tools
+# M01 <String> ; Displays <String> and waits for user interaction
+TOOL_CHANGE = '''G77      ; Move to release position
+M10 O6.0 ; Stop spindle
+M01 Insert tool TOOL
+G76      ; Move to reference point to ensure correct coordinates after tool change
+M10 O6.1 ; Start spindel'''
+
+
+# gCode finishing the program
+POSTAMBLE = '''G77      ; Move to release position
+M10 O6.0 ; Stop spindle'''
+
+
+# gCode header with information about CAD-software, post-processor and date/time
+HEADER = ''';Exported by FreeCAD
+;Post Processor: {}
+;CAM file: {}
+;Output Time: {}
+'''.format(__name__, FreeCAD.ActiveDocument.FileName, str(datetime.datetime.now()))
+
+
+def export(objectslist, filename, argstring):
+
+  gcode = HEADER
+
+  for obj in objectslist:
+
+    for command in obj.Path.Commands:
+
+      # Manipulate tool change commands

```

Comment:
You should remove many of the empty white lines that you left after removing the comments so that the code is more compact.